enterprise_access/apps/api/v1/views.py

```diff
@@ -134,8 +167,153 @@ def _validate_subsidy_request(self):
             logger.exception(error_msg)
             raise SubsidyRequestCreationError(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY)
 
+    def _validate_subscription_plan_uuid(self, subscription_plan_uuid):
+        """
+        Args:
+            subsidy_request_uuid: a valid uuid
+        Raises:
+            SubsidyRequestError: if subscription_plan_uuid not present or invalid
+        """
+        if not subscription_plan_uuid:
+            error_msg = (
+                'You must provide a License Subscription UUID so subsidy requests '
+                'can be completed (so a License can be assigned to the user).'
+            )
+            logger.exception(error_msg)
+            raise SubsidyRequestError(error_msg, status.HTTP_400_BAD_REQUEST)
+
+        try:
+            validate_uuid(subscription_plan_uuid)
+        except ParseError as exc:
+            error_msg = f'Subscription Plan UUID provided ({subscription_plan_uuid}) is not a valid UUID'
+            logger.exception(error_msg)
+            raise SubsidyRequestError(error_msg, status.HTTP_400_BAD_REQUEST) from exc
+
+    def _verify_subsidies_remaining(self, subscription_plan_uuid, subsidy_request_uuids):
+        """
+        Check license manager to make sure there are licenses remaining.
+        """
+        try:
+            client = LicenseManagerApiClient()
+            subscription_overview = client.get_subscription_overview(subscription_plan_uuid)
+        except (ConnectionError, HTTPError, Timeout) as exc:
+            error_msg = (
+                'We were unable to process your request at this time. '
```

Comment:
Maybe something more specific to leave ourselves a better breadcrumb when/if this happens?  Perhaps unable to assign (code/licenses) at this time. Please try again later or reach out ..... blah blah blah? Thoughts?

Reply:
yeah that sounds reasonable

Reply:
```                
'We were unable to approve/deny the selected coupon requests at this time. '
'Please try again, or reach out to customer support if you '
'continue to experience issues. '
```

and then replace coupon with license for the other 👍

Reply:
Perhaps "Please try again in a little while, or ...

Alena, ideas on language for this error?